core/src/main/scala/scalaz/BindRec.scala

```diff
@@ -47,9 +49,40 @@ trait BindRec[F[_]] extends Bind[F] { self =>
   val bindRecSyntax = new scalaz.syntax.BindRecSyntax[F] { def F = BindRec.this }
 }
 
+trait MonadRecSafely[M[_], A] {
+  def apply[S[_]: BindRec: Applicative](lift: M <~> S): S[A]
```

Comment:
Is `BindRec` constraint intentional here (as opposed to just `Bind`)? Isn't this supposed to represent an operation implemented against the `Monad` interface, which is potentially unsafe if the monad is not lazy? On that note, if it represents a potentially unsafe operation, `MonadRecSafely` doesn't seem like a suitable name.

Reply:
It's intentional. It is representing an unsafe operation, but only ones which can be transformed into safe ones.